include/usb/usb_ch9.h

```diff
@@ -0,0 +1,246 @@
+/** USB Standard Request Codes */
+#define USB_REQ_GET_STATUS		0x00
+#define USB_REQ_CLEAR_FEATURE		0x01
+#define USB_REQ_SET_FEATURE		0x03
+#define USB_REQ_SET_ADDRESS		0x05
+#define USB_REQ_GET_DESCRIPTOR		0x06
+#define USB_REQ_SET_DESCRIPTOR		0x07
+#define USB_REQ_GET_CONFIGURATION	0x08
+#define USB_REQ_SET_CONFIGURATION	0x09
+#define USB_REQ_GET_INTERFACE		0x0A
+#define USB_REQ_SET_INTERFACE		0x0B
+#define USB_REQ_SYNCH_FRAME		0x0C
+
+/** USB Standard Feature Selectors */
+#define USB_SFS_ENDPOINT_HALT		0x00
+#define USB_SFS_REMOTE_WAKEUP		0x01
+#define USB_SFS_TEST_MODE		0x02
+
+/** Bits that are to be used for USB_REQ_GET_STATUS request */
+#define USB_GET_STATUS_SELF_POWERED	BIT(0)
+#define USB_GET_STATUS_REMOTE_WAKEUP	BIT(1)
+
+static inline bool usb_reqtype_is_to_host(struct usb_setup_packet *setup)
+{
+	return setup->RequestType.type == USB_REQTYPE_DIR_TO_DEVICE ?
+		true : false;
+}
+
+static inline bool usb_reqtype_is_to_device(struct usb_setup_packet *setup)
+{
+	return setup->RequestType.type == USB_REQTYPE_DIR_TO_HOST ?
+		true : false;
+}
+
+/** Header of an USB descriptor */
+struct usb_desc_header {
+	uint8_t bLength;
+	uint8_t bDescriptorType;
+} __packed;
+
+/** USB Standard Device Descriptor */
+struct usb_device_descriptor {
+	uint8_t bLength;
+	uint8_t bDescriptorType;
+	uint16_t bcdUSB;
+	uint8_t bDeviceClass;
+	uint8_t bDeviceSubClass;
+	uint8_t bDeviceProtocol;
+	uint8_t bMaxPacketSize0;
+	uint16_t idVendor;
+	uint16_t idProduct;
+	uint16_t bcdDevice;
+	uint8_t iManufacturer;
+	uint8_t iProduct;
+	uint8_t iSerialNumber;
+	uint8_t bNumConfigurations;
+} __packed;
+
+/** USB Unicode (UTF16LE) String Descriptor */
+struct usb_string_descriptor {
+	uint8_t bLength;
+	uint8_t bDescriptorType;
+	uint16_t bString;
+} __packed;
+
+/** USB Association Descriptor */
+struct usb_association_descriptor {
+	uint8_t bLength;
+	uint8_t bDescriptorType;
+	uint8_t bFirstInterface;
+	uint8_t bInterfaceCount;
+	uint8_t bFunctionClass;
+	uint8_t bFunctionSubClass;
+	uint8_t bFunctionProtocol;
+	uint8_t iFunction;
+} __packed;
+
+/** USB Standard Configuration Descriptor */
+struct usb_cfg_descriptor {
+	uint8_t bLength;
+	uint8_t bDescriptorType;
+	uint16_t wTotalLength;
+	uint8_t bNumInterfaces;
+	uint8_t bConfigurationValue;
+	uint8_t iConfiguration;
+	uint8_t bmAttributes;
+	uint8_t bMaxPower;
+} __packed;
+
+/** USB Standard Interface Descriptor */
+struct usb_if_descriptor {
+	uint8_t bLength;
+	uint8_t bDescriptorType;
+	uint8_t bInterfaceNumber;
+	uint8_t bAlternateSetting;
+	uint8_t bNumEndpoints;
+	uint8_t bInterfaceClass;
+	uint8_t bInterfaceSubClass;
+	uint8_t bInterfaceProtocol;
+	uint8_t iInterface;
+} __packed;
+
+/** USB Standard Endpoint Descriptor */
+struct usb_ep_descriptor {
+	uint8_t bLength;
+	uint8_t bDescriptorType;
+	uint8_t bEndpointAddress;
+	uint8_t bmAttributes;
+	uint16_t wMaxPacketSize;
+	uint8_t bInterval;
+} __packed;
+
+/** Descriptor Types defined in Specification Table 9-6 */
+#define USB_DESCR_DEVICE		1
```

Comment:
I personally dislike `DESCR`. I would much rather use `USB_DESC_DEVICE` and so on

Reply:
I have used DESCR because we have it already in the tree, but there is probably more DESC als DESCR so I can change it if you insist :smile:

Reply:
@carlescufi fixed in https://github.com/zephyrproject-rtos/zephyr/pull/27045/commits/b196528bcd395ef9fc14165bcf065195c0b4f167